internal/framework/services/s3management/data_source_s3_accesskey.go

```diff
@@ -0,0 +1,108 @@
+package s3management
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services/s3management"
+	s3managementService "github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services/s3management"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+)
+
+var _ datasource.DataSourceWithConfigure = (*accessKeyDataSource)(nil)
+
+// NewAccesskeyDataSource creates a new data source for the accesskey resource.
+func NewAccesskeyDataSource() datasource.DataSource {
+	return &accessKeyDataSource{}
+}
+
+type accessKeyDataSource struct {
+	client *services.SdkBundle
+}
+
+// Metadata returns the metadata for the data source.
+func (d *accessKeyDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_s3_accesskey"
+}
+
+// Configure configures the data source.
+func (d *accessKeyDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	// Add a nil check when handling ProviderData because Terraform
+	// sets that data after it calls the ConfigureProvider RPC.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*services.SdkBundle)
```

Comment:
maybe here we can save only the client that we are interested in the resources and not the bundle.

Reply:
done